BraintreeCore/src/main/java/com/braintreepayments/api/AnalyticsEventParams.kt

```diff
@@ -0,0 +1,18 @@
+package com.braintreepayments.api
+
+public data class AnalyticsEventParams private constructor(
+    val payPalContextId: String?,
+    val linkType: String?,
+    val isVaultRequest: Boolean,
+) {
+    class Builder {
```

Comment:
Is this inner builder class needed? Could we instead use the `AnalyticsEventParams`'s constructor wherever we need to create new instances?

I feel that the builder pattern is more of a Java pattern where there isn't any support for named parameters or default values.

Reply:
Got it, and I understand that the goal is to migrate everything to Kotlin. However, creating this object is to avoid having to change too many files whenever we add a new parameter to the events. The problem we have now is that we have both Java and Kotlin. This design pattern would help us create an instance without having to specify all the parameters in Java.

For example, in Java, when I only need to specify isVaultRequest, I have to do something like this:
`AnalyticsEventParams(null, null, true)`

With builder patter:
`AnalyticsEventParams.Builder.setIsVaultRequest`

The purpose of all this is that if we increase the number of parameters, we won't have to modify so many classes.

Reply:
Updated here: [930bef6](https://github.com/braintree/braintree_android/pull/1031/commits/930bef6fd4fef8eb137784283e87d9ac294b6b3b)

Reply:
@richherrera IMO, using default values would be better for this use-case.
please check https://kotlinlang.org/docs/functions.html#default-arguments.
>fun foo(
    bar: Int = 0,
    baz: Int,
) { /*...*/ }

>foo(baz = 1) // The default value bar = 0 is used

```
data class AnalyticsEventParams constructor(
    var payPalContextId: String? = null,
    var linkType: String? = null,
    var isVaultRequest: Boolean = false,
)
```
You could use any combination of the parameters and it'd work.
Add @JvmOverloads for java compatibility.

Reply:
@saperi22 we looked into this there was one hiccup. Since there are no named params in Java the String param becomes ambiguous when it's `(payPalContextId, isVaultRequest)` vs. `(linkType, isVaultRequest)`. The compiler complains because it can't determine if the `String` in the `AnalyticsEventParams(String, boolean)` parameter tuple is a `payPalContextId` or a `linkType` without additional information.